### ISSUE TITLE
Diffing files while symlinking is broken

### ISSUE THREAD

**Issue by UrsMetz** [technicalpickles/homesick#129]
When linking a castle and a symlink needs to override an existing file the user is asked what he wants to do. When choosing `d` for diffing the following error occurs (using Homesick 1.1.1):
```
/home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/utils.rb:153:in `block in collision_accepted?': undefined local variable or method `source' for #<Homesick::CLI:0x007f2d59b09ae0> (NameError)
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/shell/basic.rb:254:in `block in file_collision'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/shell/basic.rb:237:in `loop'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/shell/basic.rb:237:in `file_collision'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/utils.rb:153:in `collision_accepted?'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/actions/file_actions.rb:78:in `handle_symlink_action'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/actions/file_actions.rb:61:in `ln_s'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/utils.rb:201:in `block in symlink_each'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/utils.rb:188:in `block (2 levels) in each_file'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/utils.rb:162:in `each'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/utils.rb:162:in `block in each_file'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/actions.rb:184:in `block in inside'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/2.1.0/fileutils.rb:125:in `chdir'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/2.1.0/fileutils.rb:125:in `cd'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/actions.rb:184:in `inside'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/utils.rb:158:in `each_file'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/utils.rb:200:in `symlink_each'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/cli.rb:131:in `block (2 levels) in link'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/cli.rb:130:in `each'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/cli.rb:130:in `block in link'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/actions.rb:184:in `block in inside'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/2.1.0/fileutils.rb:125:in `chdir'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/2.1.0/fileutils.rb:125:in `cd'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/actions.rb:184:in `inside'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/lib/homesick/cli.rb:123:in `link'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
	from /home/urs/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/homesick-1.1.1/bin/homesick:9:in `<top (required)>'
	from /home/urs/.rbenv/versions/2.1.5/bin/homesick:23:in `load'
	from /home/urs/.rbenv/versions/2.1.5/bin/homesick:23:in `<main>'
```

**Comment by JCook21** [comment]
I think this is a duplicate of #134. This should be fixed in #146.

**Comment by JCook21** [comment]
I've merged #146 to fix this and will try to cut a new release with this in over the next few days.
